macapp/App/Sources/App/MonitoringFeature.swift

```diff
@@ -40,9 +40,12 @@ extension MonitoringFeature.RootReducer {
   func reduce(into state: inout State, action: Action) -> Effect<Action> {
     switch action {
 
-    case .loadedPersistentState(.some(let persistent)):
-      guard persistent.resumeOnboarding == nil else { return .none }
-      return self.configureMonitoring(current: persistent.user, previous: nil)
+    case .loadedPersistentState(.some(let stored)):
+      guard stored.resumeOnboarding == nil ||
+        stored.resumeOnboarding == .checkingFullDiskAccessPermission(upgrade: true) else {
```

Comment:
this line is what fixes the `v2.7.0/1` bug, almost the whole rest of the PR is just the tests to faithfully simulate and capture the regression